### PR TITLE
Week2

### PR DESCRIPTION
**Pull request by alaashhadat**

### CODE REVIEW

File: week2/homework/src/addtodo.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const fs = require('fs');
+const { promisify } = require('util');
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+
+async function addToList(...command) {
+  let list = await readFile('todolist.json', 'utf8');
+  list = JSON.parse(list);
+  list.push(`\n${command.join(' ')}`);
```
Comment (eidosam) [inline week2/homework/src/addtodo.js]: I don' think you need the new line `\n` preceding the item.

File: week2/homework/src/addtodo.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const fs = require('fs');
+const { promisify } = require('util');
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+
+async function addToList(...command) {
```
Comment (eidosam) [inline week2/homework/src/addtodo.js]: Maybe `todoItem` instead of `command`?

File: week2/homework/src/index.js [hunk 3]
```diff
@@ -1,3 +1,41 @@
...
+const addToList = require('./addtodo');
+const removeToDo = require('./removetodo');
+const reset = require('./reset');
+const update = require('./update');
+const helpCommand = require('./help');
+function actions(commandsCallFunc) {
+  commandsCallFunc.then(function() {
+    return renderList();
+  });
+}
```
Comment (eidosam) [inline week2/homework/src/index.js]: Neat!
But also don't forget to handle the errors, so `.catch(// ...`

File: week2/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,41 @@
...
+
+if (command === 'list') {
+  renderList();
+}
+ else if (command === 'add') {
+  actions(addToList(...argument));
+}
+ else if (command === 'remove') {
+  if (argument < 0) {
```
Comment (eidosam) [inline week2/homework/src/index.js]: 0 shouldn't be accepted as well, so `if (argument <= 0) { // ...`

File: week2/homework/src/index.js [hunk 5]
```diff
@@ -1,3 +1,41 @@
...
+    console.log('index must be greater than 0');
+  }
+  actions(removeToDo(argument));
```
Comment (eidosam) [inline week2/homework/src/index.js]: Maybe you don't want to call the function if doesn't meet your condition? So `else { actions(removeToDo(argument)); // .. `

File: week2/homework/src/index.js (hunk 3 above)
Comment (alaashhadat) [reply week2/homework/src/index.js]: thank you :)

File: week2/homework/src/index.js (hunk 4 above)
Comment (alaashhadat) [reply week2/homework/src/index.js]: thank you I'll change it

**Comment by wouterkleijn** [comment]
Thanks @eidosam for the comments. And thank you @alaashhadat for addressing them! @eidosam any further comments or this one is done?

**Pull request closed without merging**